### PR TITLE
PHPUnit: add OptionsTest

### PR DESCRIPTION
**Pull request by aaemnnosttv**
## Summary

This PR can be summarized in the following changelog entry:

* PHPUnit: add OptionsTest

## Relevant technical choices

* `test_*` methods apply to both MS and single site contexts
* `test_network_mode_*` are MS-specific with the plugin in a network mode context

## Checklist:
- [x] My code is tested and passes existing unit tests.
- [x] My code has an appropriate set of unit tests which all pass.
- [x] My code is backward-compatible with WordPress 4.7 and PHP 5.4.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [x] My code has proper inline documentation.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

### CODE REVIEW

File: tests/phpunit/integration/Core/Storage/Google/OptionsTest.php [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+		$this->assertFalse( $options->get( 'test_option' ) );
+		update_option( 'test_option', 'test-value' );
+
+		$this->assertEquals( 'test-value', $options->get( 'test_option' ) );
+	}
+
+	/**
+	 * @group ms-required
+	 */
+	public function test_network_mode_get() {
```
Comment (felixarntz) [inline tests/phpunit/integration/Core/Storage/Google/OptionsTest.php]: Tests annotated like that should also start with a `if ( ! is_multisite() )` condition that skips the respective test.

Comment (aaemnnosttv) [reply tests/phpunit/integration/Core/Storage/Google/OptionsTest.php]: This is handled automatically by `\WP_UnitTestCase_Base::checkRequirements`.

See https://github.com/WordPress/wordpress-develop/blob/ed1233d05f0ea12df0683dcbf38734aa69e39269/tests/phpunit/includes/abstract-testcase.php#L773-L781

Comment (aaemnnosttv) [reply tests/phpunit/integration/Core/Storage/Google/OptionsTest.php]: These groups should be skipped automatically by [`\WP_UnitTestCase_Base::checkRequirements`](https://github.com/WordPress/wordpress-develop/blob/ed1233d05f0ea12df0683dcbf38734aa69e39269/tests/phpunit/includes/abstract-testcase.php#L773-L781). It worked when I tested in multisite mode locally.

I'll take a closer look and see what's going on there.

Comment (felixarntz) [reply tests/phpunit/integration/Core/Storage/Google/OptionsTest.php]: The latest fix is great, polyfilling 5.1 - awesome!

**Pull request merged**